### DISCUSSION TITLE
How to request API Gateway endpoint using PHP AWS SDK?

### DISCUSSION THREAD

**Discussion by megusta420** [aws/aws-sdk-php#2421] [Q&A] [answered]
I have `https://api-id.execute-api.region.amazonaws.com/stage` url endpoint, how to query it using SDK?

Found `ApiGatewayClient` but can't figure out how to use it properly and didn't find any helpful usage examples on the internet.

Some methods require some ConnectionID or another arguments, I am sure there should be a simple way.

For instance my Api url is `https://myawesomeid.execute-api.eu-central-1.amazonaws.com/prod/` where "myawesomeid" is my ApiId.

**Comment by yenfryherrerafeliz** [comment]
Hi @megusta420, thank you for reaching out and sorry for the delayed response. Just to clarify, your question is if is there any way to execute a request to an end-point defined with the API Gateway service, by using the SDK?

**Comment by megusta420** [comment, marked as answer]
@yenfryherrerafeliz hello and yes, sure! I've already implemented it by using a simple Guzzle call, but maybe there is a way to get instantiated HTTP client from SDK to not create it manually. Or to request it by SDK but as far as I understood, SDK is not to query API Gateway endpoints.

**Reply by yenfryherrerafeliz** [reply]
Hi @megusta420. Basically your understanding is right. At this time the SDK does not have a functionality to consume API Gateway endpoints through an API Gateway client.

[a comment by an automated account is not shown]
